validator/src/micro.rs

```diff
@@ -84,75 +84,103 @@ impl<TValidatorNetwork: ValidatorNetwork + 'static> NextProduceMicroBlockEvent<T
         let mut expected_next_ts;
 
         let return_value = loop {
+            // Wait for the expected timestamp to arrive before producing the block.
+            // We sleep at the beginning of the loop such that we can `continue` into the sleep.
+            // We can't sleep while holding the blockchain lock.
+            if delay > Duration::default() {
```

Comment:
```suggestion
            if delay.is_zero() {
```
I like `delay.is_zero()` better here, as it is in my opinion a more meaningful expression.

If yielding is not a problem the conditional could also be removed, sleeping unconditionally albeit for a zero duration. I did not check if that is problematic or not and I just assume that `sleep(Duration::ZERO).await;` will in fact yield.